.github/workflows/build-plugin-zip.yml

```diff
@@ -6,18 +6,111 @@ on:
+            if [[ $OLD_VERSION == *"rc"* ]]; then
+              NEW_VERSION=$(npx semver $OLD_VERSION -i prerelease)
+            else
+              # WordPress version guidelines: If minor is 9, bump major instead.
+              IFS='.' read -r -a OLD_VERSION_ARRAY <<< "$OLD_VERSION"
+              if [[ ${OLD_VERSION_ARRAY[1]} == "9" ]]; then
+                NEW_VERSION="$(npx semver $OLD_VERSION -i major)-rc.1"
+              else
+                NEW_VERSION="$(npx semver $OLD_VERSION -i minor)-rc.1"
+              fi
+            fi
+          fi
+          echo "::set-output name=new_version::$(echo $NEW_VERSION)"
+          IFS='.' read -r -a NEW_VERSION_ARRAY <<< "$NEW_VERSION"
+          RELEASE_BRANCH="release/${NEW_VERSION_ARRAY[0]}.${NEW_VERSION_ARRAY[1]}"
+          echo "::set-output name=release_branch::$(echo $RELEASE_BRANCH)"
+
+      - name: Configure git user name and email
+        run: |
+          git config user.name "Gutenberg Repository Automation"
+          git config user.email gutenberg@wordpress.org
+
+      - name: Create and switch to release branch
+        if: |
+          github.event.inputs.version == 'rc' &&
+          ! contains( steps.get_version.outputs.old_version, 'rc' )
+        run: git checkout -b "${{ steps.get_version.outputs.release_branch }}"
+
+      - name: Switch to release branch
+        if: |
+          github.event.inputs.version == 'stable' ||
+          contains( steps.get_version.outputs.old_version, 'rc' )
+        run: |
+          git fetch --depth=1 origin "${{ steps.get_version.outputs.release_branch }}"
+          git checkout "${{ steps.get_version.outputs.release_branch }}"
+
+      - name: Update plugin version
+        env:
+          VERSION: ${{ steps.get_version.outputs.new_version }}
+        run: |
+          cat <<< $(jq --tab --arg version "${VERSION}" '.version = $version' package.json) > package.json
+          cat <<< $(jq --tab --arg version "${VERSION}" '.version = $version' package-lock.json) > package-lock.json
+          sed -i "s/${{ steps.get_version.outputs.old_version }}/${VERSION}/g" gutenberg.php
+          sed -i "s/${{ steps.get_version.outputs.old_version }}/${VERSION}/g" readme.txt
+
+      - name: Commit the version bump
+        run: |
+          git add gutenberg.php package.json package-lock.json readme.txt
+          git commit -m "Bump plugin version to ${{ steps.get_version.outputs.new_version }}"
+          git push --set-upstream origin "${{ steps.get_version.outputs.release_branch }}"
+
+      - name: Cherry-pick to trunk
+        run: |
+          git checkout trunk
+          TRUNK_VERSION=$(jq --raw-output '.version' package.json)
+          if [[ ${{ steps.get_version.outputs.old_version }} == "$TRUNK_VERSION" ]]; then
+            git cherry-pick "${{ steps.get_version.outputs.release_branch }}"
+            git push
+          fi
+
+
   build:
     name: Build Release Artifact
     runs-on: ubuntu-latest
+    needs: bump-version
```

Comment:
Won't this prevent the job from running on PRs since bump-version doesn't run there?

Reply:
The `if: always()` on the line right below this makes sure it's also run if the `bump-version` job isn't. (You can verify by checking the "Build Release Artifact" job for this PR.)